test/rebar_test_utils.erl

```diff
@@ -259,6 +260,10 @@ top_level_deps([{{Name, Vsn, Ref}, _} | Deps]) ->
 %%% Helpers %%%
 %%%%%%%%%%%%%%%
 check_results(AppDir, Expected, ProfileRun) ->
+    State = rebar_state:new(),
```

Comment:
yeah this is the kind of forced invocation into other callsites we should be able to avoid by not requiring `State` weaved through for standard projects.

Reply:
The call sites are conflicting with the new ones (e.g. all_apps/2 exists in both)